agent/src/test/java/com/appland/appmap/record/RecorderTest.java

```diff
@@ -98,6 +112,71 @@ public void testUnwriteableOutputFile() throws IOException {
     assertTrue(exception.toString().indexOf("java.lang.RuntimeException: ") == 0);
   }
 
+  @Test
+  // for Files.move when: REPLACE_EXISTING, ATOMIC_MOVE
+  public void testWriteFileAcrossFilesystems() throws IOException {
+    //  /dev/shm exists only on Linux
+    assumeThat(System.getProperty("os.name"), is("Linux"));
+    systemErrRule.clearLog();
+    String sourceFilename = "/tmp/recordertest_file";
+    String targetFilename = "/dev/shm/recordertest_file";
+    File sourceFile = new File(sourceFilename);
+    File targetFile = new File(targetFilename);
+    Exception exception = null;
+    // if the file exists createNewFile returns false
+    sourceFile.createNewFile();
+
+    // Copying a file across filesystems should not throw an exception.
+    final Recording recording = new Recording("recorderName", sourceFile);
+    try {
+      recording.moveTo(targetFilename);
+    } catch (RuntimeException e) {
+      exception = e;
+      System.out.println(exception.getMessage());
```

Comment:
This is ok for now. but if you say `e.printStackTrace()`, it will do the right thing (i.e. print the message and stack trace to `System.err`).